mlrun/feature_store/steps.py

```diff
@@ -304,6 +406,16 @@ def do(self, event):
             event[self._get_key_name(part, self.timestamp_col)] = extracted_part
         return event
 
+    def _do_pandas(self, event):
+        timestamp = self._extract_timestamp(event)
+        # Extract specified parts
```

Comment:
Please add more comments like these, its impossible to review logic for long methods without sitting hours and following the code. See how I write code in the `mlrun.frameworks` or the `decorrator` you just reworked.